### PR TITLE
backporting - applying patch form HIVE-5181

### PR DESCRIPTION
**Pull request by jhsenjaliya**

### CODE REVIEW

**Comment by jhsenjaliya** [comment]
I have build the hive 12 rpm with this change: vcc-hive-0.12.0-2.0.0-201401100318.x86_64.rpm     using this build job: https://jenkins.service.altiscale.com/view/Hadoop/job/hive-buildrpm-0.12.0-Jay/

I have tested this rpm on desktop as well.

File: metastore/src/java/org/apache/hadoop/hive/metastore/RawStore.java [hunk 1]
```diff
@@ -68,11 +80,13 @@
    *
    * @return true or false
    */
+  @CanNotRetry
```
Comment (ashahab-altiscale) [inline metastore/src/java/org/apache/hadoop/hive/metastore/RawStore.java:83]: So if I understand this, commit and rollback won't be retried?

Comment (jhsenjaliya) [inline metastore/src/java/org/apache/hadoop/hive/metastore/RawStore.java:83]: yes, in specific case, when there is a logical failure in RawStore

**Comment by chutong-altiscale** [comment]
Checked and it matches what is in for HIVE-5181.

**Pull request merged**